drivers/mtd/mtd_config_fs.c

```diff
@@ -0,0 +1,2494 @@
+      return -ENOENT;
+    }
+
+  do
+    {
+      rd_addr = fs->step_addr;
+      rc = nvs_prev_ate(fs, &(fs->step_addr), &step_ate);
+      if (rc)
+        {
+          return rc;
+        }
+
+      if (step_ate.len && (nvs_ate_valid(fs, &step_ate))
+          && step_ate.id != NVS_SPECIAL_ATE_ID
+          && step_ate.expired == CONFIG_MTD_CONFIG_ERASEDVALUE)
+        {
+          break;
+        }
+    }
+  while (fs->step_addr != fs->ate_wra);
+
+  if (fs->step_addr == fs->ate_wra)
+    {
+      return -ENOENT;
+    }
+
+#ifdef CONFIG_MTD_CONFIG_NAMED
+  rc = nvs_flash_rd(fs, (rd_addr & ADDR_BLOCK_MASK) + step_ate.offset,
+                    key, step_ate.key_len);
+  if (rc)
+    {
+      ferr("Key read failed, rc=%d\n", rc);
+      return rc;
+    }
+#else
+  rc = nvs_flash_rd(fs, (rd_addr & ADDR_BLOCK_MASK) + step_ate.offset,
+                    key, step_ate.key_len);
+  if (rc)
+    {
+      ferr("Key read failed, rc=%d\n", rc);
+      return rc;
+    }
+
+  memcpy(pdata->id, key, sizeof(pdata->id));
+  memcpy(pdata->instance, key + sizeof(pdata->id), sizeof(pdata->instance));
+#endif
+
+  rc = nvs_flash_rd(fs, (rd_addr & ADDR_BLOCK_MASK) + step_ate.offset +
+                    step_ate.key_len, pdata->configdata, step_ate.len);
+  if (rc)
+    {
+      ferr("Value read failed, rc=%d\n", rc);
+      return rc;
+    }
+
+  *len = step_ate.len;
+  return OK;
+}
+
+/****************************************************************************
+ * Name: mtdconfig_open
+ ****************************************************************************/
+
+static int mtdconfig_open(FAR struct file *filep)
+{
+  return OK;
+}
+
+/****************************************************************************
+ * Name: mtdconfig_close
+ ****************************************************************************/
+
+static int mtdconfig_close(FAR struct file *filep)
+{
+  return OK;
+}
+
+/****************************************************************************
+ * Name: mtdconfig_read
+ ****************************************************************************/
+
+static ssize_t mtdconfig_read(FAR struct file *filep, FAR char *buffer,
+                           size_t buflen)
+{
+  return -ENOTSUP;
+}
+
+/****************************************************************************
+ * Name: mtdconfig_ioctl
+ ****************************************************************************/
+
+static int mtdconfig_ioctl(FAR struct file *filep, int cmd,
+                           unsigned long arg)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct nvs_fs *fs = (FAR struct nvs_fs *)inode->i_private;
+  FAR struct config_data_s *pdata = (FAR struct config_data_s *)arg;
+  int ret;
+
+  switch (cmd)
+    {
+      case CFGDIOC_GETCONFIG:
+
+        /* Read a nvs item */
+
+        ret = nxmutex_lock(&fs->nvs_lock);
+        if (ret < 0)
+          {
+            return ret;
+          }
+
+        ret = nvs_read(fs, pdata, pdata->len);
```

Comment:
original NVS has the param len.
I keep it this way so that NVS implementation won't need changing. 
And I think it is okay to leave NVS code unchanged, it is better if NVS has some upgrades we need to merge.

Reply:
since nvs_read can access the same data through pdata->len, it's strange to pass it as an argument again.

Reply:
Alright, I will remove pdata->len param